### PR TITLE
feat(oracle connector): add possible to configure and use SSL

### PR DESCRIPTION
**Pull request by kjellwinblad**
This is not tested since I have not managed to set up SSL in the oracle docker XE image. It should be tested at least manually. Perhpas QA could help with that?

Fixes:
https://emqx.atlassian.net/browse/EMQX-12288

Release version: v/e5.?

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [x] Schema changes are backward compatible

### CODE REVIEW

File: apps/emqx_oracle/src/emqx_oracle.erl [hunk 1]
```diff
@@ -108,12 +108,21 @@ on_start(
         {timeout, ?OPT_TIMEOUT},
         {app_name, "EMQX Data To Oracle Database Action"}
     ],
+    Options1 =
+        case emqx_utils_maps:deep_get([ssl, enable], Config, false) of
+            true ->
+                SSLOptsMap = maps:get(ssl, Config),
+                SSLOpts = maps:to_list(SSLOptsMap),
+                [{ssl, SSLOpts} | Options0];
```
Comment (thalesmg) [inline apps/emqx_oracle/src/emqx_oracle.erl:116]: Nit: could `emqx_tls_lib:to_client_opts` be of use here?

**Comment by kjellwinblad** [comment]
The driver does not seem to work without a patch of OTP. There is patches for OTP 25 but not for OTP 26: https://github.com/erlangbureau/jamdb_oracle/issues/98#issuecomment-2150320459

**Comment by kjellwinblad** [comment]
I have tried the following patch

https://github.com/erlangbureau/jamdb_oracle/blob/master/test/ssl-10.8-otp-25.patch

On OTP versions:

OTP-25.2.3
OTP-25.3.2.12

These don't work. `ssl:connect` get stuck after the patch has been applied. The following command gets stuck:

```
ssl:connect("oracle-test-instance.cerkngbiw6l9.eu-west-1.rds.amazonaws.com", 2484, [{verify, verify_none}]).
```

**Comment by id** [comment]
We decided to reject this one.

**Pull request closed without merging**